Don't create symlink for Pandoc when it's not being vendored.

https://github.com/conda-forge/quarto-feedstock/pull/36#issuecomment-1946923128

IMPORTANT: This patch is applied upstream, but not yet released on 1.4.x. This
patch should be removed with the next 1.4.x release. It's only good for 1.4.550
and earlier.

Original commit:

commit 75264062555182478fa37e45040b6cad8e1cec21
Author: Charles Teague <cteague@gmail.com>
Date:   Thu Feb 15 13:47:49 2024 -0500

    quarto-bld - pandoc symlink vendoring
    
    Note that if we are being instructed not to vendor binaries,
    Pandoc won't be present in the architecture specific directory, so
    just skip this step.
    
    See https://github.com/conda-forge/quarto-feedstock/pull/36

diff --git a/package/src/common/configure.ts b/package/src/common/configure.ts
index e5f83603e..277e9d20e 100644
--- a/package/src/common/configure.ts
+++ b/package/src/common/configure.ts
@@ -46,7 +46,15 @@ export async function configure(
   copyQuartoScript(config, config.directoryInfo.bin);
 
   info("Creating architecture specific Pandoc link");
-  copyPandocScript(config, join(config.directoryInfo.bin, "tools"));
+  const vendor = Deno.env.get("QUARTO_VENDOR_BINARIES");
+  if (vendor === undefined || vendor === "true") {
+    // Quarto tools may look right in the bin/tools directory for Pandoc
+    // so make a symlink that points to the architecture specific version.
+    // Note that if we are being instructed not to vendor binaries,
+    // Pandoc won't be present in the architecture specific directory, so 
+    // just skip this step.
+    copyPandocScript(config, join(config.directoryInfo.bin, "tools"));
+  }
 
   // record dev config. These are versions as defined in the root configuration file.
   const devConfig = createDevConfig(
